backend/src/controller/listing.ts

```diff
@@ -3,46 +3,139 @@ import ExampleListingList from "../constants/ExampleListingList";
 import ExampleUserList from "../constants/ExampleUserList";
 import productsArray from "../controller/product";
 
-const fullInfoListings = new Array();
-ExampleListingList.map((listing) => {
-  const listingItem = productsArray.find(
-    (product) => product.productId === listing.productID
-  );
-  fullInfoListings.push({
-    listingID: listing.listingID,
-    price: listing.price,
-    amount: listing.amount,
-    highestBid: listing.highestBid,
-    startDate: listing.startDate,
-    endDate: listing.endDate,
-    likes: listing.likes,
-    productImageUrl: listingItem.productImageUrl,
-    productName: listingItem.productName,
-    productDescription: listingItem.productDescription,
-    productID: listingItem.productId,
+const listingList = ExampleListingList;
+
+let fullInfoListings = new Array();
+function fillFullInfo() {
+  fullInfoListings = new Array();
+  listingList.map((listing) => {
+    const listingItem = productsArray.find(
+      (product) => product.productId === listing.productID
+    );
+    fullInfoListings.push({
+      listingID: listing.listingID,
+      userID: listing.userID,
+      price: listing.price,
+      amount: listing.amount,
+      highestBid: listing.highestBid,
+      startDate: listing.startDate,
+      endDate: listing.endDate,
+      likes: listing.likes,
+      productImageUrl: listingItem.productImageUrl,
+      productName: listingItem.productName,
+      productDescription: listingItem.productDescription,
+      productID: listingItem.productId,
+    });
   });
-});
+}
+fillFullInfo();
 
 module.exports.getAllListings = (req: Request, res: Response) => {
-  res.json(ExampleListingList);
+  ExampleListingList
+    ? res.json(ExampleListingList)
```

Comment:
suggestion: check for length, since an empty array is truthy. But i don't think you need the check at all, since you're populating the array, no? Also, shouldn't this be `fullInfoListing` instead of `ExampleListingList`?

Reply:
Or is the ExampleListingList a different thing from fullInfoListing?

Reply:
ExampleListingList is different from fullIntoListing. object in ExampleListingList has only basic props like userID, productID, price, amount. Object in fullInfoListing has joined product`s things, like image, name etc. I prepared this exactly for our ProductCard component. Thanks for advice about length of array